inspire_disambiguation/core/ml/models.py

```diff
@@ -672,6 +675,26 @@ def fit(self, n_jobs=8):
         )
         self.clusterer.fit(self.X, self.y)
 
+    def score(self, test_uuids, labels):
+        """
+        Return the clustering statistics (b3 precision, b3 recall, b3 f1 score)
+        and wrongly clustered samples for training, test and the whole dataset.
+
+        Args:
+            test_uuids - list of signatures uuids used for testing
+            labels - list of labels (author id) for test dataset
+        """
+        all_uuids = np.vectorize(lambda x: x.signature_uuid)(self.X).flatten()
+        test_uuids_array = np.array(test_uuids)
+        mask = np.array([item in test_uuids_array for item in all_uuids])
```

Comment:
as I've already said in the past, a `list`/`array` is not the right data structure for testing membership. You need something like a `dict` or a `set` which has `O(1)` membership test.

Reply:
ok, then you probably want to use [`np.isin`](https://docs.scipy.org/doc/numpy-1.13.0/reference/generated/numpy.isin.html#numpy.isin). In general you shouldn't go back and forth between numpy/python as that defeats the whole point of using numpy (which is performance).